orchestrator/search/retrieval/builder.py

```diff
@@ -0,0 +1,76 @@
+from sqlalchemy import Select, Table, and_, select
+from sqlalchemy_utils.types.ltree import Ltree
+
+from orchestrator.db.models import AiSearchIndex
+from orchestrator.search.filters import FilterSet, LtreeFilter
+from orchestrator.search.schemas.parameters import BaseSearchParameters
+
+
+class QueryBuilder:
```

Comment:
A bit nitpicky, but: this technically doesn't need to be a class, can be done with plain functions (you're not really using `self`).